src/DropDown/DropDown.component.tsx

```diff
@@ -207,15 +207,18 @@ export class DropDown extends React.Component<DropDownProps> {
         const mouseEnter$ = fromEvent(dropDown, 'mouseenter');
         const mouseLeave$ = fromEvent(dropDown, 'mouseleave');
 
-        this.clickedSub = merge(
+        this.clickedSub$ = merge(
             outsideClick$.pipe(mapTo(false)),
             escapeKey$.pipe(mapTo(false)),
             dropDownClick$.pipe(map(() => !this.state.clicked))
         ).subscribe(clicked => {
             this.setState({ clicked });
+            if (this.props.onTriggered) {
```

Comment:
I think this may fire unnecessarily, at times. e.g., the dropdown is closed, but the user hits the `esc` key.

In fact, this method would have issues when using the `both` or `hover` triggers, as it doesn't account for the `trigger` prop or how the two values combine the for `both` (e.g. you click to keep it opened, but hover and unhover).

I'd probably recommend `useEffect` but this isn't a functional component :/

Reply:
I think this can be an optimization. I did notice that in those cases the event fires and the caret isn't predictable, but the way MegaMenu is presently wired this will work.

Reply:
But the second point- the behavior is broken for `both` and `hover`. Unless you're saying we can hold off on fixing it until we need it for more than just `click`

Reply:
The latter.